query/src/sql/planner/binder/project.rs

```diff
@@ -121,8 +127,17 @@ impl Binder {
                         column_name: expr_name,
                         index,
                         data_type,
-                        scalar: Some(Box::new(bound_expr)),
+                        scalar: Some(Box::new(bound_expr.clone())),
                     };
+                    if has_order_by
+                        && !matches!(bound_expr, Scalar::BoundColumnRef(BoundColumnRef { .. }))
+                    {
+                        output_context
+                            .order_by_columns
+                            .as_mut()
+                            .unwrap()
```

Comment:
maybe return result::err is better?

Reply:
In fact, `order_by_columns` is always `Some`, because it's initialized by `Option` with empty vector, so `unwrap` won't panic

Reply:
It's better to handle the unwrap(Though it may not happen) here :D